### DISCUSSION TITLE
Connecting GUI to running server

### DISCUSSION THREAD

**Discussion by axelbr** [gama-platform/gama#59] [Q&A] [answered]
Hi!
I just wanted to know if there is a way to connect the GAMA gui to a running headless server.
Moreover, is there a way to retrieve any graphical outputs (such as maps or images) from the headless server?
Thank you.

**Comment by lesquoyb** [comment, marked as answer]
Hello Axel,

There is no way to directly connect the GAMA gui to a GAMA-server yet. There has been some libraries developed to ease the interaction with GAMA-server that you can find [here](https://github.com/gama-platform/gama.clients), do not hesitate to ask for improvements/development if you have specific needs.

To retrieve the graphical outputs you have many possibilities, I recommend that you try to use the new [snapshot](https://gama-platform.org/wiki/next/OperatorsSZ#snapshot) operator that makes it possible to save a given display with the resolution you want into a file. There is an example model of how to use in the GAMA plugin library under the `Images` directory:
![image](https://user-images.githubusercontent.com/6374469/236594553-72e6124e-726e-4dbc-b4ea-52cddbd3ce3b.png)

**Reply by axelbr** [reply]
Ok, thanks for the info.

Thank you for the hint, I will try it out!

**Reply by axelbr** [reply]
Hello, I have a follow-up question: is there a way to capture snapshots on headless servers too? Snapshots do not work when running headless.

**Reply by lesquoyb** [reply]
really ? if you have an example of a model with non functional snapshot can you share it ? I'll open an issue about it.
You can try the `autosave` facet of your display, there is an example in the model `issue3570`:
![image](https://user-images.githubusercontent.com/6374469/237057291-8785815f-e41d-4327-b5c3-62ab7b0eac6d.png)

**Reply by axelbr** [reply]
ok, I figured it out in the meantime. Rendering via snapshots in server mode (not legacy) is only possible for Gama versions 1.9.1 and up. I had the v1.9.0 docker image.

**Reply by axelbr** [reply]
See here:

https://github.com/gama-platform/gama/issues/3689
